internal/services/v1/experimental.go

```diff
@@ -98,12 +99,14 @@ func NewExperimentalServer(dispatch dispatch.Dispatcher, permServerConfig Permis
 				grpcvalidate.UnaryServerInterceptor(),
 				handwrittenvalidation.UnaryServerInterceptor,
 				usagemetrics.UnaryServerInterceptor(),
+				perfinsights.UnaryServerInterceptor(permServerConfig.PerformanceInsightMetricsEnabled),
```

Comment:
I understand the convenience, but this would be an annoying extra bit to the call stack, plus it's just computation we don't need to do, even if it's negligible.

Consider adding a helper to make the plumbing more palatable, if you explore that avenue

Reply:
The system returns a no-op interceptor if it isn't enabled, which is the standard gRPC way of doing this

Reply:
fair